agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -4517,6 +4535,139 @@ bool backhaul_manager::handle_slave_channel_selection_response(ieee1905_1::CmduM
+        }
+
+        send_cmdu_to_bus(cmdu_tx, controller_bridge_mac, bridge_info.mac);
+
+        return false;
+    }
+
+    /*
+        TODO: BACKHAUL_STA_STEERING can be accepted in wired backhaul too.
+              Code below is incorrect in that case.
+    */
+    auto active_hal = get_wireless_hal();
+    if (!active_hal) {
+        LOG(ERROR) << "Couldn't get active HAL";
+        return false;
+    }
+
+    auto bssid = bh_sta_steering_req->target_bssid();
+
+    auto associate = active_hal->roam(bssid, channel);
+    if (!associate) {
+        LOG(ERROR) << "Couldn't associate active HAL with bssid: " << bssid;
+
+        LOG(DEBUG) << "Sending ACK message to the originator, mid=" << std::hex << mid;
+        send_cmdu_to_bus(cmdu_tx, controller_bridge_mac, bridge_info.mac);
+
+        auto response = create_backhaul_steering_response(
+            wfa_map::tlvErrorCode::eReasonCode::
+                BACKHAUL_STEERING_REQUEST_REJECTED_TARGET_BSS_SIGNAL_NOT_SUITABLE);
+
+        if (!response) {
+            LOG(ERROR) << "Failed to build Backhaul STA Steering Response message.";
+            return false;
+        }
+
+        send_cmdu_to_bus(cmdu_tx, controller_bridge_mac, bridge_info.mac);
+
+        return false;
+    }
+
+    m_backhaul_sta_steering_enable = true;
+
+    return true;
+}
+
+bool backhaul_manager::create_backhaul_steering_response(
+    const wfa_map::tlvErrorCode::eReasonCode &error_code)
+{
+    auto cmdu_tx_header =
+        cmdu_tx.create(0, ieee1905_1::eMessageType::BACKHAUL_STEERING_RESPONSE_MESSAGE);
+    if (!cmdu_tx_header) {
+        LOG(ERROR) << "Failed to create Backhoul STA Steering Response message";
+        return false;
+    }
+
+    auto bh_steering_resp_tlv = cmdu_tx.addClass<wfa_map::tlvBackhaulSteeringResponse>();
+    if (!bh_steering_resp_tlv) {
+        LOG(ERROR) << "Couldn't addClass<wfa_map::tlvBackhaulSteeringResponse>";
+        return false;
+    }
+
+    auto active_hal = get_wireless_hal();
+    if (!active_hal) {
+        LOG(ERROR) << "Couldn't get active HAL";
+        return false;
+    }
+
+    sMacAddr sta_mac;
+    auto interface = active_hal->get_iface_name();
+    get_iface_mac(interface, sta_mac);
+
+    LOG(DEBUG) << "Interface: " << interface << "MAC: " << sta_mac;
+
+    bh_steering_resp_tlv->target_bssid()         = tlvf::mac_from_string(active_hal->get_bssid());
+    bh_steering_resp_tlv->backhaul_station_mac() = sta_mac;
+
+    if (!error_code) {
+        bh_steering_resp_tlv->result_code() =
+            wfa_map::tlvBackhaulSteeringResponse::eResultCode::SUCCESS;
+    } else {
+        bh_steering_resp_tlv->result_code() =
+            wfa_map::tlvBackhaulSteeringResponse::eResultCode::FAILURE;
+    }
+
+    auto error_code_tlv = cmdu_tx.addClass<wfa_map::tlvErrorCode>();
+    if (!bh_steering_resp_tlv) {
+        LOG(ERROR) << "Couldn't addClass<wfa_map::tlvErrorCode>";
+        return false;
+    }
+
+    error_code_tlv->reason_code() = error_code;
```

Comment:
This should only be added if error_code != 0.

Reply:
fixed